src/Symfony/Component/DependencyInjection/Argument/ServiceLocator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class ServiceLocator
```

Comment:
@jvasseur @unkind you're missing eg exceptions - the wording about them in PSR-11 is a requirement here also. ArrayAccess is not an interface - it's just an magical incantation for the engine.

> to use this component outside the symfony framework I would need to : require an psr-11 container

???: this *would be* an implementation of PSR-11, so you'd need nothing else really.